### ISSUE TITLE
[HOLD for payment 2024-01-17] [$500] Distance - Able to request money by distance by adding a stop same as finish point without enterring start point From

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29895]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.86.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @rakshitjain13
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1697639775327799

## Action Performed:
Action Performed:
1. Open app
2. Click on '+' FAB icon in LHS
3. Click on Request Money
4. Select Distance tab
5. Click on Add stop and fill India
6. Click on Stop and fill India
7. Click on Next button to proceed further
8. Able to Create Request

## Expected Result:
It should not create request with not valid waypoints

## Actual Result:
 Able to create distance request with duplicate waypoints

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/633981e9-1809-4c40-bd9a-cc49d651f9f3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

https://github.com/Expensify/App/assets/93399543/08d274f0-6e1c-4420-af21-d1a46f000984

</details>

<details>
<summary>iOS: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/689f84d5-6b74-411f-ba77-32e580cd3b9b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/bad19b9b-e733-4552-84dc-2ee167413a77

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/2826f26f-f291-4b7d-b93b-9671cc93a898

https://github.com/Expensify/App/assets/93399543/7f6fceaa-656f-43ad-8e58-47f8c517c941

</details>

<details>
<summary>MacOS: Desktop</summary>

 

https://github.com/Expensify/App/assets/93399543/6956ba72-a2ef-42ee-ae18-9617905c994d

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f177f4854fa6146e</li>
        <li>Upwork Job ID: 1714703219702423552</li>
        <li>Last Price Increase: 2023-11-01</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Contributor | 28035387</li>
</ul></li>
    </ul>
</details>

**Comment by rakshitjain13** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Next Button Activated with Same Stop and Finish Point in Request Money
### What is the root cause of that problem?
In ```src/components/DistanceRequest.js```
we check for validwaypoint and then decide wether we have to go on next step or not . The logic is to check the size of validatedWaypoints array which is shown below .
https://github.com/Expensify/App/blob/6cd9c84cba8ac9d0a55573e7118192478c6913df/src/components/DistanceRequest/index.js#L183-L191
Now to generate validatedWaypoints the code is in file```Src/libs/TransactionUtils.js```
https://github.com/Expensify/App/blob/6cd9c84cba8ac9d0a55573e7118192478c6913df/src/libs/TransactionUtils.ts#L416-L445
This function fails in a edge case where starting waypoints are invalid because here lastWaypointIndex is equal to -1 in starting which suggests that first waypoint will always be valid when waypoints size > 1 .
### What changes do you think we should make in order to solve the problem?
We have to have increment lastWaypointIndex till we find some valid waypoint and then it is good to go . Here is my solution

```

function getValidWaypoints(waypoints: WaypointCollection, reArrangeIndexes = false): WaypointCollection { 
     const sortedIndexes = Object.keys(waypoints).map(getWaypointIndex).sort(); 
     const waypointValues = sortedIndexes.map((index) => waypoints[`waypoint${index}`]); 
     // Ensure the number of waypoints is between 2 and 25 
     if (waypointValues.length < 2 || waypointValues.length > 25) { 
         return {}; 
     } 
  
     let lastWaypointIndex = -1; 
     let foundFirstValidWaypoint = false;
  
     return waypointValues.reduce<WaypointCollection>((acc, currentWaypoint, index) => { 
         const previousWaypoint = waypointValues[lastWaypointIndex]; 

           // Skip all starting invalid waypoints
            if (foundFirstValidWaypoint == false && !waypointHasValidAddress(currentWaypoint)) {
                lastWaypointIndex += 1;
                return acc;
            } else {
                foundFirstValidWaypoint = true;
            }

  
         // Check if the waypoint has a valid address 
         if (!waypointHasValidAddress(currentWaypoint)) { 
             return acc; 
         } 
  
         // Check for adjacent waypoints with the same address 
         if (previousWaypoint && currentWaypoint?.address === previousWaypoint.address) { 
             return acc; 
         } 
  
         const validatedWaypoints: WaypointCollection = {...acc, [`waypoint${reArrangeIndexes ? lastWaypointIndex + 1 : index}`]: currentWaypoint}; 
  
         lastWaypointIndex += 1; 
  
         return validatedWaypoints; 
     }, {}); 
 } 
```
Here I have introduced a new variable named as foundFirstValidWaypoint which is false in starting and once we find a validwaypoint we will set it to true till the time we will increment lastWaypointIndex in each iteration .

### What alternative solutions did you explore? (Optional)
NA

**Comment by kbecciv** [comment]
Linked PR https://github.com/Expensify/App/issues/27045

**Comment by strepanier03** [comment]
I'm not able to recreate this following the steps in the report. 

I see an error:

<img width="638" alt="image" src="https://github.com/Expensify/App/assets/10925636/303d9a85-03c2-4dda-9bb0-b486ccff3904">

Checking the link PR shared above, this GH is a regression. 

Anyone able to recreate this still?

**Comment by strepanier03** [comment]
Doubled checked and both staging and prod show the error.

**Comment by rakshitjain13** [comment]
@strepanier03 I think you misread the steps. You can follow one of the videos for information.

**Comment by DylanDylann** [comment]
I think My PR https://github.com/Expensify/App/pull/28308 will fix this bug too

**Comment by rakshitjain13** [comment]
@DylanDylann I don't think so as in the above case validatedWaypoints size will be 2

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Distance - Able to request money by distance by adding a stop same as finish point without enterring start point From

### What is the root cause of that problem?
Let's see the logic to validate waypoints
https://github.com/Expensify/App/blob/732d48e23ac132b1f91fcd004d6941d1b0237c34/src/libs/TransactionUtils.ts#L424-L444

The logic to update lastWaypointIndex in the end of function
https://github.com/Expensify/App/blob/732d48e23ac132b1f91fcd004d6941d1b0237c34/src/libs/TransactionUtils.ts#L441

Then if the condition here is true
https://github.com/Expensify/App/blob/732d48e23ac132b1f91fcd004d6941d1b0237c34/src/libs/TransactionUtils.ts#L430
lastWaypointIndex will be not updated. It causes this bug
### What changes do you think we should make in order to solve the problem?
I suggest we should remove lastWaypointIndex and use index-1 instead 

### What alternative solutions did you explore? (Optional)
we also can move this line 
https://github.com/Expensify/App/blob/732d48e23ac132b1f91fcd004d6941d1b0237c34/src/libs/TransactionUtils.ts#L441
before the condition 
https://github.com/Expensify/App/blob/732d48e23ac132b1f91fcd004d6941d1b0237c34/src/libs/TransactionUtils.ts#L430

**Comment by strepanier03** [comment]
After retesting I see where I messed up the steps and was able to create the expense.

**Comment by strepanier03** [comment]
@rushatgabhane friendly bump on the proposal [here](https://github.com/Expensify/App/issues/29895#issuecomment-1772076972) when you get a chance.

**Comment by strepanier03** [comment]
@rushatgabhane - Do you have the bandwidth to review soon? Is there something else we should be doing with this?

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

We are able to create two consecutive duplicate waypoints when there are `stops` added to the waypoints. So, we are trying to solve the problem by implementing the [comment here](https://github.com/Expensify/App/issues/29895#issuecomment-1792016995)

### What is the root cause of that problem?

Currently, the logic supports consecutive duplicate waypoints when there are no `stops` added to the waypoints (i.e when only two waypoints are present). Support for consecutive duplicate waypoints with additional `stops` in waypoints is not there. That is the root cause.

### What changes do you think we should make in order to solve the problem?

In the logic of determining `validatedWaypoints` [here](https://github.com/Expensify/App/blob/aa30ef2ea44279637eebb2517b4e08ce88f90e7b/src/libs/TransactionUtils.ts#L429C1-L432C10), we already check if we have two consecutive duplicate waypoints and ignore such duplicates here. This means that the length of `waypoints` and `validatedWaypoints` will be different when there are duplicates. We can use this information to solve this problem.

1) First, let us ensure that on `submitWaypoints` [here](https://github.com/Expensify/App/blob/aa30ef2ea44279637eebb2517b4e08ce88f90e7b/src/components/DistanceRequest/index.js#L185), we will prevent the user to go to the next page by comparing the lengths of `waypoints` and `validatedWaypoints` like this:

        if (_.size(validatedWaypoints) < 2 || _.size(validatedWaypoints) !== _.keys(waypoints).length || hasRouteError || isLoadingRoute || isLoading) {
            setHasError(true);
            return;
        }

2) Now, since we already have `hasError` set, let us ensure that the rendering also happen by replacing our condition [here](https://github.com/Expensify/App/blob/aa30ef2ea44279637eebb2517b4e08ce88f90e7b/src/components/DistanceRequest/index.js#L225) like this:

                {((hasError && (_.size(validatedWaypoints) < 2) || _.size(validatedWaypoints) !== _.keys(waypoints).length) || hasRouteError) && (

3) Finally, we need to set the error message for duplicates [here](https://github.com/Expensify/App/blob/aa30ef2ea44279637eebb2517b4e08ce88f90e7b/src/components/DistanceRequest/index.js#L157) like this:

        if (_.keys(waypoints).length > 2 &&  _.size(validatedWaypoints) !== _.keys(waypoints).length) {
            return {0: translate('iou.error.duplicateWaypointsErrorMessage')};
        }

### What alternative solutions did you explore? (Optional)

**Comment by strepanier03** [comment]
All good Melvin, things are moving forward.

**Comment by rakshitjain13** [comment]
> I think we can simplify the solution.
> 
> 1. We should always have a `start` and an `end point`
> 2. No two consecutive waypoints should be same. (This allows for A -> B -> and back to A)
> 
> What do you think? Any edge cases?

Adding the start as compulsory can solve the problem as our getValidatefunction fails in that case only. Thus adding this solution in my proposal as optional https://github.com/Expensify/App/issues/29895#issuecomment-1769064322

**Comment by rushatgabhane** [comment]
c+ reviewed 🎀 👀 🎀

**Comment by DylanDylann** [comment]
We have [another issue](https://github.com/Expensify/App/issues/29886) with new UX behavior: we only display add button if user fill 2 waypoints. It will make this bug can't reproduce more

**Comment by paultsimura** [comment]
@rushatgabhane the info @DylanDylann mentioned above was also discussed with you here: https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1698999067236989 

Was there any further discussion about the "add stop" button functionality that led to proceeding with this issue?

**Comment by rojiphil** [comment]
> I think we can simplify the solution.
> 
> 1. We should always have a `start` and an `end point`
> 2. No two consecutive waypoints should be same. (This allows for A -> B -> and back to A)

@rushatgabhane 

Few queries I have:
A) Which specific scenario do you think in (1) mentioned above is not addressed by [my proposal here](https://github.com/Expensify/App/issues/29895#issuecomment-1792954251)? It would be great if you share the test steps for this.
B) Does the [approved proposal here](https://github.com/Expensify/App/issues/29895#issuecomment-1796001122) handle the scenario (A->B->B->A)? I think this scenario is part of (2) mentioned above.
Or am I missing something obvious here? 
Please clarify

**Comment by rushatgabhane** [comment]
@rojiphil sorry i missed your proposal

**Comment by rojiphil** [comment]
> A) Which specific scenario do you think in (1) mentioned above is not addressed by [my proposal here](https://github.com/Expensify/App/issues/29895#issuecomment-1792954251)? Can you please share the test steps for this?

@rushatgabhane 

Oh ok! I notice the scenario for (A). No need to share the steps. This happens if the start is not entered.
A peculiar observation here is that when `start` and `finish` destination is not entered and `stop` destination is attempted, it will populate the `finish` destination. Also, an empty `stop` destination is displayed. This does not look right.
I think, either
i) The add `stop` destination should be disabled until `start` and `finish` is entered OR
ii)The add `stop` destination should fill in the empty `start` or `finish` destination (in the same order) and prevent an empty `stop` destination to be shown. 
I think, based on your comment in [slack here](https://expensify.slack.com/archives/C01GTK53T8Q/p1699004801417899?thread_ts=1698999067.236989&cid=C01GTK53T8Q), it looks like (ii) is the way to go.
Does this make sense?

**Comment by rojiphil** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/29895#issuecomment-1792954251)

> ii)The add `stop` destination should fill in the empty `start` or `finish` destination (in the same order) and prevent an empty `stop` destination to be shown.

I have updated the proposal with the required changes for the above additional objective. 
Also, the remaining part of the proposal still holds good. It addresses A->B->B->C scenario. 

> We have https://github.com/Expensify/App/issues/29886 with new UX behavior: we only display add button if user fill 2 waypoints. It will make this bug can't reproduce more

Also,  if handling of `add` button display is managed by the other issue, we can restrict our scope to handling the duplicate waypoint scenario addressed here in my proposal.

**Comment by rojiphil** [comment]
Hmmm... I should have read through the other issue also [here](https://github.com/Expensify/App/issues/29886#issuecomment-1794541149).
 Looks like there is a consensus on showing `Add` button only after `Start` and `Finish` are populated. 
I think that would leave us with the pending issue of handling duplicates.

**Comment by rakshitjain13** [comment]
I think handling duplicate is not necessary as Google Maps also allows to have duplicate addresses. Also, duplicate addresses are handled by getValidWaypoints.

**Comment by rojiphil** [comment]
> duplicate addresses are handled by getValidWaypoints.

While I agree that `getValidWaypoints` will ignore the consecutive duplicate addresses, I think some additional work is required to display the message for duplicate entries (i.e. in A->B->B->A scenarios) and also to prevent the usage of `next` to move ahead. no?

**Comment by rakshitjain13** [comment]
<img width="410" alt="image" src="https://github.com/Expensify/App/assets/54314936/30be1f5f-31a2-4e19-879f-030139b6a9d0">

I think the current implementation suffices the above scenario without any error messages cc: @rojiphil

**Comment by rojiphil** [comment]
@rushatgabhane 
What are your thoughts on addressing the consecutive duplicate endpoints scenario of A->B->B->A?

**Comment by danieldoglas** [comment]
@strepanier03 can you maybe take this discussion RE https://github.com/Expensify/App/issues/29895#issuecomment-1800974078 to product?

**Comment by rushatgabhane** [comment]
> What are your thoughts on addressing the consecutive duplicate endpoints scenario of A->B->B->A?

A->B->B->A should not be a valid endpoint sequence.
I think any two consecutive endpoints shouldn't be same

**Comment by rushatgabhane** [comment]
We should also make the start field mandatory regardless of the button disable / enable.
What I mean is, even without the disable UX the validate function should be correct when the user clicks next.

I hope this makes sense. What do you think? @rojiphil

**Comment by rojiphil** [comment]
> We should also make the start field mandatory regardless of the button disable / enable. What I mean is, even without the disable UX the `validate()` function should be correct when the user clicks next.
> 
> I hope this makes sense. What do you think? @rojiphil

@rushatgabhane 
I still do not understand why we need to make the start field mandatory as the appropriate message is displayed on click of `Next`. Sorry if I am missing something simple here. 
Can you please share a failure test video when we do not make the start field mandatory?

Also, I noticed that the latest version has some changes for the implementation. 
Here is the latest test video. 

https://github.com/Expensify/App/assets/3004659/b932db4e-10b9-48bd-8dc7-5e00e67d14dd

**Comment by rojiphil** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/29895#issuecomment-1792954251)

Have integrated changes to adjust according to the recent changes to [PR here](https://github.com/Expensify/App/pull/30548)
Also, have updated the logic and added test video.
Kindly review.

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Users can go to the next page when entering two consecutive same endpoints

### What is the root cause of that problem?
We don't have logic to prevent user from clicking next button when they enter two consecutive same endpoints
### What changes do you think we should make in order to solve the problem?
Note that in `getValidWaypoints` function we had logic to remove a waypoint if 2 consecutive endpoints is same
In here 
https://github.com/Expensify/App/blob/bf2d545b5e2d40d7e72a51983112ef9fa9daf3f7/src/components/DistanceRequest/index.js#L180
we should check if `_.size(validatedWaypoints) < _.size(waypoints)` we will prevent user from submitting 

Also need to update here
https://github.com/Expensify/App/blob/bf2d545b5e2d40d7e72a51983112ef9fa9daf3f7/src/components/DistanceRequest/index.js#L220

In this function https://github.com/Expensify/App/blob/bf2d545b5e2d40d7e72a51983112ef9fa9daf3f7/src/components/DistanceRequest/index.js#L147
we also add a message like "please remove duplicated waypoints" if `_.size(validatedWaypoints) < _.size(waypoints)`  

We also prevent user going to next page by deeplink when they enter 2 consecutive same waypoints 
https://github.com/Expensify/App/blob/bf2d545b5e2d40d7e72a51983112ef9fa9daf3f7/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsPage.js#L98
In here we also need to add condition like we did in DistanceRequest Page (`_.size(validatedWaypoints) < _.size(waypoints)`)

**Comment by strepanier03** [comment]
I'm going to close this GH as the linked one above was opened before this one and the issues is resolved there.

**Comment by rushatgabhane** [comment]
@strepanier03 i think we should reopen because the validate func isn't correct https://github.com/Expensify/App/issues/29895#issuecomment-1792954251

**Comment by rojiphil** [comment]
> This issue is fixed and we removed the duplicate error context here: [#29803 (comment)](https://github.com/Expensify/App/issues/29803#issuecomment-1768881898).

@dukenv0307 
Can you please explain how is this issue fixed?
As I understand from the comment you referred to, the errors under point (2) has been addressed by the PR you mentioned. 
However, the duplicate endpoint scenario of  `A > B > B > A` is still not addressed. 
And, that is precisely what this issue is all about.

**Comment by rojiphil** [comment]
@strepanier03 @rushatgabhane 
I also think that we should reopen this. 
If we know that `A > B > B > A` is going to be considered as `A > B > A` in server, then, I think it is always better to prevent the user from entering this. I think, this will be confusing to the user.

**Comment by conorpendergrast** [comment]
Reopening based on the earlier comments!

**Comment by DylanDylann** [comment]
@rushatgabhane Please help to take a look at [my proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931)

**Comment by DylanDylann** [comment]
@rushatgabhane Please help to check [my updated proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931) that cover deeplink case

**Comment by danieldoglas** [comment]
Bumped @rushatgabhane on Slack

**Comment by strepanier03** [comment]
Thank you @danieldoglas 🙌

**Comment by rushatgabhane** [comment]
Yeah, I think we should go forward with @rojiphil's proposal https://github.com/Expensify/App/issues/29895#issuecomment-1792954251

@rojiphil  actively pushed this issue to find the root cause

**Comment by rushatgabhane** [comment]
🎀 👀 🎀

**Comment by rojiphil** [comment]
> @rojiphil's proposal doesn't cover deeplink case, user still go to next page by deeplink when they enter 2 consecutive same waypoints. Please help to check [my proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931) again.

@DylanDylann 
I will check this out tomorrow.

**Comment by rojiphil** [comment]
> @rojiphil's proposal doesn't cover deeplink case, user still go to next page by deeplink when they enter 2 consecutive same waypoints. Please help to check [my proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931) again.

@DylanDylann 
For the deeplink case, can you please share the test steps with actual and expected result on the current staging version? 
I tried reproducing this but could not. Not sure if I am missing something.
If you could also attach a test video that would be great.

**Comment by DylanDylann** [comment]
@rojiphil I updated [proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931) to detail it
cc @rushatgabhane

**Comment by rojiphil** [comment]
> [@rojiphil](https://github.com/rojiphil)'s proposal doesn't cover deeplink case, user still go to next page by deeplink when they enter 2 consecutive same waypoints

@rushatgabhane 

As I understand, the approach to the primary solution in @DylanDylann proposal is same as my proposal. However, an additional edge case is identified where the user comes via deep link to the `participants` page. [Here](https://github.com/Expensify/App/blob/276b0071735c056fb22a211a2bd5754486f0fdb2/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsPage.js#L69), we check if there are at least two waypoints to continue further which makes sense as it would be useless to proceed further without having at least two waypoints. Unlike this, I think it is not critical to consider duplicates for scenarios with >2 waypoints as BE handles it gracefully.But, I think it may be better to include this for consistency. The same condition that is used in my proposal can be used here. 
What do you think about this? Include or do not include?

If the decision is to be included, I would happily share $125 from my compensation to @DylanDylann for his contribution.

**Comment by DylanDylann** [comment]
@rushatgabhane 

> @rojiphil actively pushed this issue to find the root cause
I greatly appreciate his support and his participation in the ongoing discussion about the expected behavior.
But my proposal is the best one to cover all cases and give more value. There are many issues where the completed proposal is selected instead of the proposal that only fixing one case

**Comment by rojiphil** [comment]
>  I greatly appreciate his support and his participation in the ongoing discussion about the expected behavior.
> But my proposal is the best one to cover all cases and give more value. There are many issues where the completed proposal is selected instead of the proposal that only fixing one case

@DylanDylann 

While I thank you for your appreciation towards finding the root cause, I think my involvement is not just limited to discussions.
I actively [proposed](https://github.com/Expensify/App/issues/29895#issuecomment-1792954251) a solution with 4 distinct steps which was presented before your [proposal](https://github.com/Expensify/App/issues/29895#issuecomment-1807988931).
And your proposal consists of the same 4 distinct steps with a slight change in condition. Not sure of the additional value that is brought here.
However, I do agree that you brought forth the deep link case. But, here, I think it is not a critical implementation in the context of this issue as we are dealing with >2 waypoint scenarios. 

It is understandable If the solution to an issue is simple and another proposal with additional non-critical fixes gets selected. But, curious to know if there are any issues where a large part of the proposal is taken from another proposal and it gets approved with a non-critical value added on top of it. I think, normally there would be a separate GH issue for additional fixes.

Nevertheless, every issue is different. So, I would leave it to @rushatgabhane to conclude on this.

**Comment by DylanDylann** [comment]
@rojiphil

**Comment by strepanier03** [comment]
@rushatgabhane little bump when you get the chance to weigh in on the above.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by rushatgabhane** [comment]
@danieldoglas i think we should go with @rojiphil's proposal because the edge case could have been caught in a PR review

**Comment by rushatgabhane** [comment]
@danieldoglas @strepanier03 could you please add `waiting for copy` label.

We need a copy for an error message when duplicate waypoints are entered. 

"Please remove duplicate waypoints"

**Comment by danieldoglas** [comment]
This was merged today, should be deployed to staging soon.

**Comment by rushatgabhane** [comment]
1. The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/28618

2. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/28618/files#r1454558927

3. A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N.A.

4. Determine if we should create a regression test for this bug. Sure!

5. If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again 
        1. Click Request Money via Global FAB(+)
        2. Select Start Address (A)
        3. Select Finish Address (B)
        4. Click Add stop and add Address (B)
        5. Verify that you see a duplicate waypoint error message.

**Comment by rushatgabhane** [comment]
@strepanier03 could you please add payment summary. Thank you! 🙇

**Comment by strepanier03** [comment]
Yup, will do, today is the release day so that is the plan.

**Comment by rakshitjain13** [comment]
@strepanier03 Accepted the offer

**Comment by strepanier03** [comment]
@rakshitjain13 - After reviewing this more carefully, I'm not sure why the payment comment had your name listed. I don't see that you worked on the PR or the review for this bug. 

I am going to need to cancel the contract and we won't be able to pay it. 

If you can link me to the work you did for this bug, I can reconsider but at this point, I don't think there's any work related to this that you need to be paid for. Is that correct?

**Comment by strepanier03** [comment]
Got it thank you @0xmiroslav , looks like it got in right before we paused the program so I'll get it sorted.

**Comment by strepanier03** [comment]
@rakshitjain13 sorry about the confusion there, that's my bad. I have sent over a new offer for the right price as the original one I sent was for $500 and it should be $50.

**Comment by strepanier03** [comment]
All that's left is the manual request payment for @rushatgabhane 

Payment summary is [here](https://github.com/Expensify/App/issues/29895#issuecomment-1896683475). Closing and assigning Jason.
